lnrpc/rpc.proto

```diff
@@ -2768,3 +2783,19 @@ message BakeMacaroonResponse {
     /// The hex encoded macaroon, serialized in binary format.
     string macaroon = 1 [json_name = "macaroon"];
 }
+
+message ChannelRevenueRequest{
+    ///  The outpoint (txid:index) of the funding transaction.
+    ChannelPoint channel_point = 1;
+
+    /// Start time is the beginning of the range that revenue should be calcualted over.
+    uint64 start_time = 2 [json_name = "start_time"];
+
+    /// End time is the end of the range that revenue should be calcualted over.
+    uint64 end_time = 3 [json_name = "end_time"];
+}
+
+message ChannelRevenueResponse{
+    /// The revenue of the channel in msat.
+    int64 revenue_msat = 1 [json_name = "revenue_msat"];
+}
```

Comment:
Question about the use case: isn't it more valuable to see the revenue per pair of channels?

Reply:
Methinks the same, see this comment (copy-pasting here since still in draft): 
> One other direction that may be worth exploring is exactly how we surface this information on the RPC interface. Rather than showing a single value (total revenue) for a channel, we could instead unpack the revenue into channel pairs. This would allow the end caller to reason about the relationship between the target channel and the set of open channels, essentially: typically where is the incoming HTLC being sourced from?

Reply:
Something of this sort appeals to me?
Request channel report for channel A, from `start_time` -> `end_time`
```
ChannelReport{
  Uptime
  Lifespan
  Revenue{
     B: x msat,
     C: y msat, 
  }
}
```